lib/core/src/Cardano/Wallet.hs

```diff
@@ -2070,9 +2075,13 @@ joinStakePool ctx currentEpoch knownPools pid poolStatus wid =
 
         liftIO $ traceWith tr $ MsgIsStakeKeyRegistered isKeyReg
 
+        dep <- fmap keyDeposit $
+            withExceptT ErrJoinStakePoolNoSuchWallet
+                $ readWalletProtocolParameters @ctx @s @k ctx wid
+
         return $ if isKeyReg
-            then Join pid
-            else RegisterKeyAndJoin pid
+            then (Join pid, 0)
+            else (RegisterKeyAndJoin pid, dep)
```

Comment:
Interesting part: here we actually return some non-zero deposit

Reply:
Why is that interesting :thinking: ?

Reply:
:sweat_smile: because here happens something. Most of the PR is types fixing. I wanted to point out where actual business logic is changed/added.